# Lepidotpera_target_enrichment

This repository contains Bash scripts useful for bioinformatics analyses of target enrichment data, and R scripts used in various analyses done in papers given below. 
The bash scripts given automate various bioinformatics tasks - trimming, bwa mapping, de-novo assembly, orthology inference, etc. There is also a small snakemake workflow (Snakefile) to enable performing these tasks one after another. 

References -
1.	Joshi M., Espeland M., Huemer P., deWaard J., & Mutanen M. 2024. Species Delimitation Under Allopatry: Genomic Insights Within and Across Continents in Lepidoptera. Insect systematics and Diversity. 8(5). (https://doi.org/10.1093/isd/ixae027)
2.	Khan M., Joshi M., Espeland M., Huemer P., Lopez-Vaamonde C., Mutanen M. 2023. Patterns of speciation in a parapatric pair of Saturnia moths as revealed by target capture. Molecular Ecology. 00, 1-18. (https://onlinelibrary.wiley.com/doi/10.1111/mec.17194)
3.	Joshi M., Espeland M., DincÄƒ V., Vila R., Tahami M. S., Dietz L., Mayer C., Martin S., Dapporto L., Mutanen M. 2022. Delimiting Continuity: Comparison of Target Enrichment and ddRAD for Delineating Admixing Parapatric Melitaea Butterflies. Systematic Entomology. 1-18. ( https://doi.org/10.1111/syen.12557)
